packages/ragbits-core/src/ragbits/core/llms/base.py

```diff
@@ -80,12 +82,15 @@ async def generate_raw(
             Raw text response from LLM.
         """
         options = (self.default_options | options) if options else self.default_options
-
+        stream_output = False
+        if hasattr(prompt, "output_type"):
+            stream_output = prompt.output_type == tps.AsyncGeneratorType
```

Comment:
Does it really make sense to declare whether the response to the query should be streamed? On the face of it, it doesn't seem to me to me to be something directly connected to the content of the prompt. Instead, it seems to me that whether we want to stream or not is more of a run-time consideration - a function of things like whether the client that connected to our system supports streaming.

So for me, streaming doesn't seem like something that should be set statically in the prompt. Instead, I would see it as one of the LLMOptions.

Reply:
i was thinking about implementation of streaming as one of the `LLMOptions` however imo it is not LLM option and is related to the api rather than a model itself: LLM generates the answer token by token anyway - and we have access to the tokens - so the decision whether we want to wait for the generation to complete or read (anyway accesible pieces) on the fly is not related to the model directly

Reply:
It's related to how we want to use the model, isn't it? 

In simplest term, you model the decision as static and related to the particular prompt and to me it seems that the decision is often dynamic and related to the properties of the particular query that the system serves (e.g., the properties of the client that generated the query)

Reply:
could agree with that - however we should than rename `LLMOptions` to `LLMClientOptions` and store there also info about output_schema and json_mode indicator.

btw in a case of` json_mode=True` we should also validate the input prompt

Reply:
resolved by implementing separate method for streaming